pkg/segment/results/blockresults/blockresult.go

```diff
@@ -563,13 +564,19 @@ func (gb *GroupByBuckets) ConvertToAggregationResult(req *structs.GroupByRequest
 		}
 
 		var bucketKey interface{}
-		bucketKey, err := utils.ConvertGroupByKey([]byte(key))
-		if len(bucketKey.([]string)) == 1 {
-			bucketKey = bucketKey.([]string)[0]
-		}
-		if err != nil {
-			log.Errorf("GroupByBuckets.ConvertToAggregationResult: failed to convert group by key: %v, err: %v", key, err)
+		var err error
+		if req.IsBucketKeySeparatedByDelim {
+			bucketKey = strings.Split(key, string(utils.BYTE_TILDE))
+		} else {
+			bucketKey, err = utils.ConvertGroupByKey([]byte(key))
+			if len(bucketKey.([]string)) == 1 {
+				bucketKey = bucketKey.([]string)[0]
+			}
+			if err != nil {
+				log.Errorf("GroupByBuckets.ConvertToAggregationResult: failed to convert group by key: %v, err: %v", key, err)
+			}
```

Comment:
nit: reverse the `if`s so that you handle the `err` first, since that's how the rest of the codebase does it